verification/tagTCI.go

```diff
@@ -3,55 +3,209 @@
+
+	tciValue := make([]byte, digestLen)
+	for i := range tciValue {
+		tciValue[i] = byte(i + 1)
 	}
 
-	// TODO: When DeriveChild is implemented, call it here to add more TCIs and call TagTCI again.
+	extendTciValue := make([]byte, digestLen)
+	for i := range extendTciValue {
+		extendTciValue[i] = byte(i + 2)
+	}
+
+	// Initialize hasher
+	var hasher hash.Hash
+	if digestLen == 32 {
+		hasher = sha256.New()
+	} else if digestLen == 48 {
+		hasher = sha512.New384()
+	} else {
+		t.Error("[ERROR]: Unsupported hash algorithm used for TCI value generation")
+	}
+
+	// Get parent context TCI values for cumulative value calculation
+	parentTci, err := c.GetTaggedTCI(parentTag)
+	if err != nil {
+		t.Fatalf("Could not get tagged TCI: %v", err)
+	}
+
+	// Cross-verify parent's TCI_CUMULATIVE
+	hasher.Write(defaultTci)
+	hasher.Write(parentTci.CurrentTCI)
+	wantCumulativeTCI = hasher.Sum(nil)
+	if !bytes.Equal(parentTci.CumulativeTCI, wantCumulativeTCI) {
+		t.Errorf("Parent node's cumulative TCI %x, expected %x", parentTci.CumulativeTCI, wantCumulativeTCI)
+	}
+
+	// Derive Child context with input data, tag it and check TCI_CUMULATIVE
+	child, err := c.DeriveChild(handle, tciValue, DeriveChildFlags(MakeDefault), 0, 0)
+	if err != nil {
+		t.Fatalf("[FATAL]: Error while creating default child handle in default context: %s", err)
+	}
+
+	newHandle, err := c.TagTCI(&child.NewContextHandle, childTag)
+	if err != nil {
+		t.Fatalf("Could not tag TCI: %v", err)
+	}
+
+	childTci, err := c.GetTaggedTCI(childTag)
+	if err != nil {
+		t.Fatalf("Could not get tagged TCI: %v", err)
+	}
+
+	if !bytes.Equal(childTci.CurrentTCI, tciValue) {
+		t.Errorf("GetTaggedTCI returned current TCI %x, expected %x", childTci.CurrentTCI, tciValue)
+	}
+
+	// Check TCI_CUMULATIVE after creating child context
+	hasher.Reset()
+	hasher.Write(defaultTci)
+	hasher.Write(childTci.CurrentTCI)
+	wantCumulativeTCI = hasher.Sum(nil)
+	if !bytes.Equal(childTci.CumulativeTCI, wantCumulativeTCI) {
+		t.Errorf("Child node's cumulative TCI %x, expected %x", childTci.CumulativeTCI, wantCumulativeTCI)
+	}
+
+	// Extend TCI to child context and check TCI_CURRENT and TCI_CUMULATIVE
+	_, err = c.ExtendTCI(newHandle, extendTciValue)
+	if err != nil {
+		t.Fatalf("Could not tag TCI: %v", err)
+	}
+
+	childExtendTci, err := c.GetTaggedTCI(childTag)
+	if err != nil {
+		t.Fatalf("Could not get tagged TCI: %v", err)
+	}
+
+	if !bytes.Equal(childExtendTci.CurrentTCI, extendTciValue) {
+		t.Errorf("GetTaggedTCI returned current TCI %x, expected %x", childExtendTci.CurrentTCI, extendTciValue)
+	}
+
+	// Check TCI_CUMULATIVE after extending input to child context
+	hasher.Reset()
+	hasher.Write(childTci.CumulativeTCI)
+	hasher.Write(childExtendTci.CurrentTCI)
+	wantCumulativeTCI = hasher.Sum(nil)
+	if !bytes.Equal(childExtendTci.CumulativeTCI, wantCumulativeTCI) {
+		t.Errorf("Child node's cumulative TCI %x, expected %x", childExtendTci.CumulativeTCI, wantCumulativeTCI)
+	}
```

Comment:
This could be a 3rd test case (since it requires `ExtendTCI` support, but the earlier part of the test doesn't)

Reply:
Have split into 3rd test case